Cython/Compiler/Symtab.py

```diff
@@ -1500,6 +1509,13 @@ def find_module(self, module_name, pos, relative_level=-1):
             absolute_fallback = True
 
         module_scope = self.global_scope()
+        if module_name == "cython" and not is_relative_import:
```

Comment:
I'd move these special cases up, simplify `if relative_level is not None and relative_level > 0:` to `if is_relative_import:` and make the `cython` import cases an else-case of that.